File: Playtech/ClientApp/src/app/models/employee.ts
import { Department } from "./department";

export class Employee {
  constructor(init?: Partial<Employee>) {
    Object.assign(this, init);
  }
  public id: number;
  public name: string;
  public salary: number | null;
  public departmentId: number | null;
  public managerId: number | null;
  public department: Department;
  public manager: Employee;
}

File: Playtech/Models/Employee.cs
namespace Playtech.Models
{
    public partial class Employee
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public decimal? Salary { get; set; }
        public int? DepartmentId { get; set; }
        public int? ManagerId { get; set; }
        public Department Department { get; set; }
        public Employee Manager { get; set; }

    }
}
